sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala

```diff
@@ -495,6 +474,8 @@ private[columnar] case object DictionaryEncoding extends CompressionScheme {
       columnType.dataType match {
         case _: IntegerType =>
           val dictionaryIds = columnVector.reserveDictionaryIds(capacity)
+          val intDictionary = dictionary.map(_.asInstanceOf[Int])
```

Comment:
@viirya Could you please see [this comment](https://github.com/apache/spark/pull/18704#discussion_r138363222) to avoid unboxing? IIUC, this code still causes unboxing.

What do you think?

Reply:
I didn't notice that comment. Let me check it then. If we can't avoid, I will revert this and only remove the unused method. Thanks.

Reply:
Thanks @kiszk. I checked the comment and code and reverted this change.